src/node_api.cc

```diff
@@ -570,8 +570,7 @@ class SetterCallbackWrapper
 
   /*virtual*/
   void SetReturnValue(napi_value value) override {
-    node::FatalError("napi_set_return_value",
-      "Cannot return a value from a setter callback.");
+    assert(false);  // Cannot return a value from a setter callback
```

Comment:
Note if you implement a setter function in JavaScript (via `Object.defineProperty()`), and you return some value from the setter function, the value is simply ignored. That suggests it should be ignored here also.